gtsfm/densify/interface_patchmatchnet.py

```diff
@@ -0,0 +1,195 @@
+                        depth_ranges[key_b, 1] = max(depth_ranges[key_b, 1], b_z)
+
+                    # if both cameras are valid cameras
+                    if key_a > 0 and key_b > 0:
+                        # calculate score for track_i in the pair views (cam_a, cam_b)
+                        score_a_b = piecewise_gaussian(
+                            a_x=self._camera_centers[i_a] - w_x, b_x=self._camera_centers[i_b] - w_x
+                        )
+                        # sum up pair scores for each track_i
+                        pair_scores[key_a, key_b] += score_a_b
+                        pair_scores[key_b, key_a] = pair_scores[key_a, key_b]
+
+        # sort pair scores, for i-th row, choose the largest (num_views-1) scores, the corresponding views are selected
+        #   as (num_views-1) source views for i-th reference view.
+        pairs = np.argsort(pair_scores, axis=0)[:, -self._num_views + 1 :][:, ::-1]
+
+        # convert float depth_ranges to integers
+        depth_ranges[:, 0] = np.floor(depth_ranges[:, 0])
+        depth_ranges[:, 1] = np.ceil(depth_ranges[:, 1])
+        depth_ranges = depth_ranges.astype(np.int32)
+
+        return pairs, depth_ranges
+
+    def __len__(self) -> int:
+        """Get the number of images
+
+        Returns:
+            length of image dictionary's keys
+        """
+        return self._num_images
+
+    def __getitem__(self, index: int) -> Dict[str, Any]:
+        """Get test data to PatchmatchNet
+        produce data containing _num_views images, the first images is the reference image
+
+        Args:
+            index: index of yield test data, the reference image ID of the test data is _keys[index]
+
+        Returns:
+            Dictionary containing:
+                "idx" test image index: int
+                "imgs" source and reference images: (num_views, image_channel, image_h, image_w)
+                "proj_matrices" projection matrices: (num_views, 4, 4)
+                "depth_min" minimum depth: int
+                "depth_max" maximum depth: int
+                "filename" output filename pattern: string
+        """
+        ref_key = self._keys[index]
+        src_keys = [self._keys[src_index] for src_index in self._pairs[index]]
+
+        cam_keys = [ref_key] + src_keys
+
+        imgs: List[np.ndarray] = [[] for _ in range(self._num_stages)]
+        proj_mats: List[np.ndarray] = [[] for _ in range(self._num_stages)]
+
+        h, w, _ = self._images[self._keys[0]].value_array.shape
+        for cam_key in cam_keys:
+            img = self._images[cam_key].value_array
+            np_img = np.array(img, dtype=np.float32) / 255.0
+            np_img = cv2.resize(np_img, (w, h), interpolation=cv2.INTER_LINEAR)
+
+            h, w, _ = np_img.shape
+
+            intrinsics = self._sfm_result.get_camera(cam_key).calibration().K()
+            extrinsics = self._sfm_result.get_camera(cam_key).pose().inverse().matrix()
+            intrinsics[:2, :] /= 8.0
```

Comment:
8 seems like a magic number -- can it be `BLAH_BLAH_SCALING_FACTOR = 8` at the top?